### PR TITLE
Reorganize docs folder; isolate dev docs and user docs; integrate foam-template docs

### PR DESCRIPTION
**Pull request by infogulch**
This is a WIP PR to reorganize the docs according to the discussion in #1025. 

TODO:

- [x] Finish writing docs/user/index.md
  - [x] Add recipes list
- [x] Merge `graph-visualisation.md` + `graph-visualization.md`
- [x] Regenerate all link reference definitions
- [x] Finish centralizing dev docs
- [ ] Make sure the site still generates correctly
- [ ] Develop and document a repeatable method to transfer user docs into the `foam-template` repo

### CODE REVIEW

**Comment by infogulch** [comment]
I finished the fixups that were apparent to me and everything looks kosher. However, I'm not sure how to validate that the website still generates correctly, and I'm not sure what approach we want to take for automation.

Thoughts?

File: docs/user/getting-started/creating-new-notes.md [hunk 1]
```diff
@@ -8,7 +8,6 @@
 - You shouldn't worry too much about categorizing your notes. You can always [[search-for-notes]], and explore them using the [[graph-visualisation]].
 
 [//begin]: # "Autogenerated link references for markdown compatibility"
-[note-templates]: note-templates "Note Templates"
-[search-for-notes]: ../recipes/search-for-notes "Search for Notes"
-[graph-visualisation]: graph-visualisation "Graph Visualisation"
```
Comment (DrakeWhu) [inline docs/user/getting-started/creating-new-notes.md:13]:
```suggestion
[graph-visualization]: ../features/graph-visualisation "Graph Visualization"
```

File: docs/index.md [hunk 2]
```diff
@@ -251,11 +251,10 @@ If that sounds like something you're interested in, I'd love to have you along o
 Foam is licensed under the [MIT license](LICENSE.txt).
 
 [//begin]: # "Autogenerated link references for markdown compatibility"
-[graph-visualisation]: features/graph-visualisation.md "Graph Visualisation"
```
[the remark this answers is not in the document]
Comment (DrakeWhu) [reply docs/index.md:254]: I don't know why this was not mentioned if it was before

**Review by DrakeWhu — APPROVED** [review]
I did a quick review and all seems good, I just suggest adding the graph visualization links in two files where they seem to be missing

**Comment by infogulch** [comment]
Good callouts, I think the missing links were caused by misspelling "visualization" (with an `s` instead of a `z`) which prevented the "link reference definition" feature from auto-generating the links. I cleaned up a few other cases as well.

**Comment by infogulch** [comment]
On a side note, I tried using the "workspace janitor" tool, to redo all the references, but that just made a mess of all the reference sections that it changed. I ended up opening every file individually and saving it a couple times to get the autofix tool to work correctly.

**Review by DrakeWhu — APPROVED** [review]
All seems good now

**Comment by DrakeWhu** [comment]
I could make sure the site generates correctly. I'm not sure on how to do it but I would be glad to help

**Review by riccardoferretti — APPROVED** [review]
This looks like such a massive improvement of the docs!

**Comment by riccardoferretti** [comment]
I am thinking of already merging the PR in this state, and take the last point on a separate PR, wdyt?

(re the site, it should work without problems as we still have the `index` file)

**Comment by riccardoferretti** [comment]
@allcontributors add @infogulch for docs

**Pull request merged**